auth/server/src/main/java/com/seagates3/policy/PolicyValidator.java

```diff
@@ -229,4 +236,181 @@ boolean isPrincipalIdValid(Principal principal) {
       validateActionAndResource(List<Action> actionList,
                                 List<Resource> resourceValues,
                                 String inputBucket);
+
+protected
+ServerResponse validatePolicyElements(JSONObject jsonObject, Set<String> policyElements,
+		Set<String> statementElements) throws JSONException{
+	 ServerResponse response = null;
+	 if(jsonObject.has(JsonDocumentFields.VERSION) && jsonObject.has(JsonDocumentFields.STATEMENT)) {
+		 String versionValue = jsonObject.get(JsonDocumentFields.VERSION).toString();
+		 if (versionValue != null) {
+            if (versionValue.isEmpty()) {
+           	 response = responseGenerator.malformedPolicy("Version field cannot be empty");
+   			 LOGGER.error("Version field cannot be empty");
+   			 return response;
+
+            } else if (!versionValue.equals("2012-10-17")){
+           	 response = responseGenerator.malformedPolicy("Policy document must have version 2012-10-17 or greater.");
+           	 LOGGER.error("Policy document must have version 2012-10-17 or greater.");
+           	 return response;
+            }
+		 }
+		 if(jsonObject.get(JsonDocumentFields.STATEMENT) instanceof JSONArray) {
+			 JSONArray arr = (JSONArray) jsonObject.get(JsonDocumentFields.STATEMENT);
+			 if(arr.length()==0) {
```

Comment:
Big method so for troubleshooting ,add debug logs wherever possible

Reply:
added